doc/config.txt
source       format              STRING  lin16
source       sample_freq         INT     8000
source       scale               FLOAT   1.0f
source       dc_shift            FLOAT   0.0f
source       noise_level         FLOAT   0.0f
params       kind                STRING  fbanks
params       suffix              STRING  mel
melbanks     nbanks              INT     15
melbanks     nbanks_full         INT     -1
melbanks     lower_freq          FLOAT   0
melbanks     higher_freq         FLOAT   4000
melbanks     vector_size         INT     200
melbanks     vector_step         INT     80
melbanks     preem_coef          FLOAT   0.0
melbanks     z_mean_source       BOOL    false
plp          order               INT     12
plp          compress_fact       FLOAT   0.3333333
plp          cep_lifter          FLOAT   22
plp          cep_scale           FLOAT   10
plp          add_c0              BOOL    false
onlinenorm   estim_interval      INT     0
onlinenorm   signal_est_end      BOOL    false
onlinenorm   file                STRING  none
onlinenorm   mean_norm           BOOL    false
onlinenorm   var_norm            BOOL    false
onlinenorm   scale_to_gvar       BOOL    false
offlinenorm  sent_mean_norm      BOOL    false
offlinenorm  sent_var_norm       BOOL    false
offlinenorm  sent_std_thr        FLOAT   0.01
offlinenorm  sent_max_norm       BOOL    false
offlinenorm  sent_chmax_norm     BOOL    false
framenorm    min_floor           FLOAT   -9999.9
framenorm    shift               FLOAT   0
posteriors   system              STRING  1BT_DCT
posteriors   length              INT     31
posteriors   add_c0              BOOL    true
posteriors   hamming             BOOL    false
posteriors   suffix              STRING  lop
posteriors   bunch_size          STRING  1
posteriors   enabled             BOOL    true
posteriors   softening_func      STRING  none 0 0 0
decoder      type                STRING  stkint
decoder      wpenalty            FLOAT   -2.0
decoder      lm_scale            FLOAT   1.0
decoder      time_pruning        INT     40
decoder      mode                STRING  decode
decoder      softening_func      STRING  log 0 0 0
decoder      num_states_per_phn  INT     1
dirs         tmp                 STRING  $C/tmp
models       hmm_defs            STRING  $T/models
models       nstates             INT     3
models       gen_from_phn_list   BOOL    false
dicts        phoneme_list        STRING  ""
dicts        lexicon1            STRING  ""
dicts        lexicon2            STRING  ""
dicts        lexicon1_save_bin   BOOL    false
dicts        lexicon2_save_bin   BOOL    false
dicts        keyword_list        STRING  none
dicts        charset             STRING  eastevrope
networks     default             STRING  $C/nets/network
networks     gen_phn_loop        BOOL    false
networks     gen_kws_net         BOOL    false
networks     omit_phn            STRING  oth
labels       suffix              STRING  rec
labels       remove_path         BOOL    true
kws          default_thr         FLOAT   -10.0
kws          thresholds_file     STRING  none
gptransc     rules               STRING  none
gptransc     symbols             STRING  none
gptransc     max_variants        INT     -1
gptransc     scale_prob          BOOL    false
gptransc     prob_thr            FLOAT   -1.0
phntransc    mode                STRING  lexgpt


history.txt
 7. 2. 2006 - fixed bug in the -p switch
            - fixed bug in frame to time conversion
15. 3. 2006 - new STK library - an arbitrary language network in the HTK format can be used for decoding
                              - a structure of linear transforms can be used in parameterization
            - calculation of HTK compatible PLP features added
	    - fixed bug in incompatible Linux/Windows libraries
	    - fixed bug in processing of file lists
	    - added fast (Quicknet compatible) evaluation of exponentials


	    


config.h
/**************************************************************************
 *  copyright            : (C) 2004-2006 by Petr Schwarz & Pavel Matejka  *
 *                                        UPGM,FIT,VUT,Brno               *
 *  email                : {schwarzp,matejkap}@fit.vutbr.cz               *
 **************************************************************************
 *                                                                        *
 *  This program is free software; you can redistribute it and/or modify  *
 *  it under the terms of the GNU General Public License as published by  *
 *  the Free Software Foundation; either version 2 of the License, or     *
 *  (at your option) any later version.                                   *
 *                                                                        *
 **************************************************************************/

#ifndef CONFIG_H
#define CONFIG_H

// MEL BANKS
#define MB_SAMLEFREQ               8000
#define MB_INPUTSIZE               1000  // = 125 ms (1 ms = 8, 25 ms = 25 * 8 = 200)
#define MB_VECTORSIZE               200  // = 25 ms
#define MB_FFTVECTORSIZE            256
#define MB_FFTVECTORSIZE_2          128
#define MB_STEP                      80  // = 10 ms
#define MB_PREEMPHASIS                0
#define MB_MELBANKS_DEF              23
#define MB_LOFREQ                    64
#define MB_HIFREQ                  4000

// TRAPS + LCRC
#define BUNCH_SIZE                    5
#define BANDS_WEIGHTS_FILE_PREFIX     "/weights/band"
#define BANDS_WEIGHTS_FILE_SUFFIX     ".weights"
#define BANDS_NORMS_FILE_PREFIX       "/norms/band"
#define BANDS_NORMS_FILE_SUFFIX       ".norms"
#define BANDS_WINDOW_FILE_PREFIX      "/windows/band"
#define BANDS_WINDOW_FILE_SUFFIX      ".window"
#define BANDS_NBANDS                 21
#define MERGER_WEIGHTS_FILE           "/weights/merger.weights"
#define MERGER_NORMS_FILE             "/norms/merger.norms"

// LCRC
#define BANDS_WINDOW_FILE

// DECODER
#define DEC_CONTEXT                 23
#define DEC_WIP                     -3.7890625
#define DEC_REMOVE                  39

// MAIN
#define NPHONEMES                   40
#define TRAP_LEN                    31
#define RECBUFF_LEN                 1000

#endif
